src/tools/file.ts

```diff
@@ -0,0 +1,187 @@
+            } catch (error: any) {
+                return {
+                    content: [
+                        {
+                            type: "text",
+                            text: JSON.stringify({
+                                error: `Error: ${error?.message || "Unknown error"}`,
+                                wholeError: error,
+                            }),
+                        },
+                    ],
+                };
+            }
+        },
+    });
+};
+
+export const createFileFromUrlTool = () => {
+    return defineTool("create-file-from-url", {
+        description: "Create a file from a URL.",
+        inputSchema: z.object({
+            url: z.string().describe("The URL of the file to create"),
+            filename: z.string().optional().describe("Optional custom filename for the file"),
+        }),
+        handler: async (directus: Directus, query: any): Promise<CallToolResult> => {
+            try {
+                const { url, filename } = query;
+
+                // Fetch the file from URL
+                const response = await fetch(url);
+
+                if (!response.ok) {
+                    throw new Error(`Failed to fetch file from URL: ${response.status} ${response.statusText}`);
+                }
+
+                // Get the MIME type from the Content-Type header
+                const contentType = response.headers.get('Content-Type') || 'application/octet-stream';
+
+                // Get a reasonable filename from either the provided filename, URL, or Content-Disposition header
+                let derivedFilename = filename;
+                if (!derivedFilename) {
+                    // Try to get filename from Content-Disposition header
+                    const contentDisposition = response.headers.get('Content-Disposition');
+                    if (contentDisposition) {
+                        const filenameMatch = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/.exec(contentDisposition);
+                        if (filenameMatch && filenameMatch[1]) {
+                            derivedFilename = filenameMatch[1].replace(/['"]/g, '');
+                        }
+                    }
+
+                    // If still no filename, extract from URL
+                    if (!derivedFilename) {
+                        derivedFilename = url.split('/').pop() || 'file';
+                        // Remove URL parameters if any
+                        derivedFilename = derivedFilename.split('?')[0];
+                    }
+                }
+
+                // Get the blob with the correct MIME type
+                const blob = await response.blob();
+                const fileObj = new File([blob], derivedFilename, { type: contentType });
+
+                // Create FormData and upload
+                const formData = new FormData();
+                formData.append('file', fileObj);
+
+                const result = await directus.request(uploadFiles(formData));
```

Comment:
Directus has a native `importFile` which we can use here. Saves a bunch of file downloading and logic in here